tensordict/tensordict.py

```diff
@@ -2656,23 +2662,75 @@ def memmap_(self, prefix=None) -> TensorDictBase:
                     "memmap is not compatible with gradients, one of Tensors has requires_grad equals True"
                 )
             if isinstance(value, TensorDictBase):
-                self._tensordict[key] = value.memmap_()
+                if prefix is not None:
+                    # ensure subdirectory exists
+                    (prefix / key).mkdir(exist_ok=True)
+                    self._tensordict[key] = value.memmap_(prefix=prefix / key)
+                    torch.save(
+                        {"batch_size": value.batch_size, "device": value.device},
+                        prefix / key / "meta.pt",
+                    )
+                else:
+                    self._tensordict[key] = value.memmap_()
                 continue
-            self._tensordict[key] = MemmapTensor(value, prefix=prefix)
+            self._tensordict[key] = MemmapTensor(
```

Comment:
what happens if value is already a MemmapTensor?
Do we create a new one with a copied content?

I'm thinking: maybe if it is a MemmapTensor placed at a different location, we could raise an exception here, telling the user that this would otherwise have a weird behaviour

```
td = TensorDict({"one": MemmapTensor(..., filename="two.memmap")}, [])
td.memmap_("./")
```
ie would we expect that  `mv two.memmap one.memmap` or that `cp two.memmap one.memmap`? Or just nothing?

Reply:
Yes, the current behaviour if the element is already a `MemmapTensor` is to create a new `MemmapTensor` with copied content. Relevant logic is here:

https://github.com/pytorch-labs/tensordict/blob/1ae3066d104ed615836c8a088d0b02f93d3ee29f/tensordict/memmap.py#L226-L228

I think an error is maybe a bit restrictive, though I do think it could be good to allow user to choose between copy and move. Maybe smth like `copy_existing=False` would either move or raise an error if the current memmap is in the wrong place.

Reply:
> Maybe smth like copy_existing=False would either move or raise an error if the current memmap is in the wrong place.

copy_existing=False (default): raises an exception if already memmap and not in the right place
copy_existing=True: creates a copy at desired loc

What if it's in `/tmp`? We should just use the current location and not raise an error neither move the data right?

Reply:
This sounds good.

> What if it's in `/tmp`?

As in, what if the user calls `td.memmap_` without specifying `prefix`? I think in that case indeed, we don't care about where the files are saved, so if they already exist we can just leave them where they are. You do run the risk of the user deleting the file from elsewhere in their code, but that's technically a risk even if you copy.

Reply:
Cool! So `memmap_()` with memmap files created without specific path is a no-op (just locks the tensordict)

Reply:
I've added the `copy_existing` argument as discussed.